### PR TITLE
⚙ Show sync manager version in settings screen

### PR DESCRIPTION
**Pull request by kornelijepetak**
## What's this?
Adds a sync manager version to the settings screen.
Mostly for debugging purposes.

### Relationships
It's complicated.

## Why do we want this?
So that we can easily distinguish which sync manager the app is using at any given time.

## How is it done?
By versioning the sync managers.

### Side effects
Yes. The sync manager version is to the _side_ of the app version/build number.

## :squid: Permissions
Me and @simonrozsival

### CODE REVIEW

File: Toggl.Core.UI/ViewModels/Settings/SettingsViewModel.cs [hunk 1]
```diff
@@ -52,7 +52,9 @@ public sealed class SettingsViewModel : ViewModel
         private IThreadSafePreferences currentPreferences;
 
         public string Title { get; private set; } = Resources.Settings;
-        public string Version => $"{platformInfo.Version} ({platformInfo.BuildNumber})";
+        public string Version => $"{platformInfo.Version} ({platformInfo.BuildNumber}) {SyncManagerDesignation}";
```
Comment (heytherewill) [inline Toggl.Core.UI/ViewModels/Settings/SettingsViewModel.cs:55]: As agreed internally:

```suggestion
        public string Version => $"{platformInfo.Version} ({platformInfo.BuildNumber}/{syncManager.Version})";
```

File: Toggl.Core.UI/ViewModels/Settings/SettingsViewModel.cs [hunk 2]
```diff
@@ -52,7 +52,9 @@ public sealed class SettingsViewModel : ViewModel
...
+
+        public string SyncManagerDesignation => $"sync v{syncManager.Version}";
```
Comment (heytherewill) [inline Toggl.Core.UI/ViewModels/Settings/SettingsViewModel.cs:57]:
```suggestion
```

File: Toggl.Core/Sync/SyncManager.cs [hunk 3]
```diff
@@ -33,6 +33,8 @@ public sealed class SyncManager : ISyncManager
 
         public IObservable<Exception> Errors { get; }
 
+        public int Version => 1;
```
Comment (heytherewill) [inline Toggl.Core/Sync/SyncManager.cs:36]:
```suggestion
        public int Version { get;} = 1;
```

File: Toggl.Core/Sync/v2/SyncManagerV2.cs [hunk 4]
```diff
@@ -7,6 +7,8 @@ public class SyncManagerV2 : ISyncManager
     {
         public bool IsRunningSync => false;
 
+        public int Version => 2;
```
Comment (heytherewill) [inline Toggl.Core/Sync/v2/SyncManagerV2.cs:10]:
```suggestion
        public int Version { get; } = 2;
```

File: Toggl.Core/Sync/v2/SyncManagerSelector.cs [hunk 5]
```diff
@@ -22,7 +22,10 @@ public static class SyncManagerSelector
             Ensure.Argument.IsNotNull(oldSyncManagerCreator, nameof(oldSyncManagerCreator));
             Ensure.Argument.IsNotNull(newSyncManagerCreator, nameof(newSyncManagerCreator));
 
-            var preferences = interactorFactory.ObserveCurrentPreferences().Execute().GetAwaiter().Wait();
+            var preferences = interactorFactory.ObserveCurrentPreferences().Execute()
+                .FirstAsync()
+                .GetAwaiter()
+                .Wait();
```
Comment (heytherewill) [inline Toggl.Core/Sync/v2/SyncManagerSelector.cs:28]:
```suggestion
                .GetResult();
```

File: Toggl.Core/Sync/SyncManager.cs (hunk 3 above)
Comment (kornelijepetak) [reply Toggl.Core/Sync/SyncManager.cs:36]: Why this?

**Pull request merged**